crates/sui-types/src/effects/object_change.rs

```diff
@@ -77,20 +85,30 @@ pub enum AccumulatorOperation {
 
 #[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
 pub enum AccumulatorValue {
-    // u64 should be sufficient for coin balance.
-    U64(u64),
+    Integer(u128),
+    IntegerTuple(u128, u128),
+}
+
+/// Accumulator objects are named by an address (can be an account address or a UID)
+/// and a type tag.
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub struct AccumulatorAddress {
+    pub address: SuiAddress,
+    pub ty: TypeTag,
```

Comment:
Can you comment on why this is a `TypeTag` instead of `StructTag`? Do we plan to support native types as the type key?